contribs/drt/src/test/java/org/matsim/contrib/drt/run/examples/DrtAbortTest.java

```diff
@@ -0,0 +1,425 @@
+		config.planCalcScore().addModeParams( new ModeParams( walkAfterRejectMode ) );
+
+		config.planCalcScore().setWriteExperiencedPlans( true );
+
+//		for ( DrtConfigGroup drtCfg : MultiModeDrtConfigGroup.get(config ).getModalElements()) {
+		//relatively high max age to prevent rejections
+//			var drtRequestInsertionRetryParams = new DrtRequestInsertionRetryParams();
+//			drtRequestInsertionRetryParams.maxRequestAge = 7200;
+//			drtCfg.addParameterSet(drtRequestInsertionRetryParams);
+			// I don't know what the above does; might be useful to understand.
+//		}
+
+		MultiModeDrtConfigGroup multiModeDrtConfig = MultiModeDrtConfigGroup.get( config );
+		DrtConfigs.adjustMultiModeDrtConfig(multiModeDrtConfig, config.planCalcScore(), config.plansCalcRoute() );
+
+		Scenario scenario = DrtControlerCreator.createScenarioWithDrtRouteFactory( config );
+		ScenarioUtils.loadScenario(scenario );
+
+		Controler controler = new Controler(scenario);
+		controler.addOverridingModule(new DvrpModule() );
+		controler.addOverridingModule(new MultiModeDrtModule() );
+		controler.configureQSimComponents( DvrpQSimComponents.activateAllModes(multiModeDrtConfig ) );
+
+		if (rejectionModule){
+			controler.addOverridingModule( new DrtRejectionModule() );
+		}
+
+		controler.run();
+
+		// yy I cannot say if the expected status is useful here.  kai, apr'23
+
+		var expectedStats = RunDrtExampleIT.Stats.newBuilder()
+							 .rejectionRate(1.0)
+							 .rejections(1)
+							 .waitAverage(Double.NaN)
+							 .inVehicleTravelTimeMean(Double.NaN)
+							 .totalTravelTimeMean(Double.NaN)
+							 .build();
+
+		// I commented this line, because NaN cannot be checked (NaN == NaN always false)
+//		RunDrtExampleIT.verifyDrtCustomerStatsCloseToExpectedStats(utils.getOutputDirectory(), expectedStats);
+	}
+
+
+	private static class MyAbortHandler implements AbortHandler, MobsimEngine{
+		public static final String COMPONENT_NAME = "DrtAbortHandler";
+		private static final String delimiter = "============";
+		private final double alpha;
+		private final double beta;
+		@Inject Network network;
+		@Inject Population population;
+		@Inject MobsimTimer mobsimTimer;
+
+		TravelTime travelTime;
+		LeastCostPathCalculator router;
+
+		// TODO we would need a DRT config group here, if we don't want to hardcode the alpha and beta values
+		// DrtConfigGroup drtConfigGroup;
+
+		private InternalInterface internalInterface;
+		private final List<MobsimAgent> agents = new ArrayList<>();
+
+		List<String> drtModes = new ArrayList<>();
+		Map<String,Double> alphas = new HashMap<>();
+		Map<String,Double> betas = new HashMap<>();
+
+		@Inject MyAbortHandler(Network network, Map<String, TravelTime> travelTimeMap, Config config ){
+			travelTime = travelTimeMap.get( TransportMode.car );
+			router = new SpeedyALTFactory().createPathCalculator( network, new OnlyTimeDependentTravelDisutility( travelTime ), travelTime );
```

Comment:
This router is created with for the default network and car travel times. This may not work for all drt modes.